### DISCUSSION TITLE
Error Initializing WalletConnect SDK

### DISCUSSION THREAD

**Discussion by KittinutzMuza** [WalletConnect/walletconnect-monorepo#3255] [Developer Support] [answered]
<img width="255" alt="Screenshot 2566-07-24 at 16 48 12" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/134215736/fe9497b6-c5f5-455c-8015-63f7477863c3">
Hi I am config follow documentation but Found the error
I build with react-native expo version 48.
Follow issue I do not found any log error please suggest way to solve this problem.

**Comment by ignaciosantise** [comment]
hey @Kittinut-Muza 👋 did you follow this extra steps for Expo48 projects? https://docs.walletconnect.com/2.0/reactnative/walletconnectmodal/installation#additional-setup-for-expo-sdk-48

1. `npx expo install expo-crypto`
2. Create this file in your root project: https://github.com/WalletConnect/modal-react-native/blob/main/example/expo-crypto-shim.js
3. Add this line before any modal imports -> `import './expo-crypto-shim.js'`

**Reply by KittinutzMuza** [reply]
Yes I follow expo 48 sdk and add projectId, providerMetadata too. 
But I found this issue. ` [Error: Provider not found: ]` on init universalProvider

**Reply by ignaciosantise** [reply]
which version of [`modal-react-native`](https://github.com/WalletConnect/modal-react-native) are you using? Can you provide more info to reproduce the issue?

**Reply by KittinutzMuza** [reply, marked as answer]
@ignaciosantise thank for answer. I found how to fix it please make it wallet connect doc.
This issue should be found in IOS simulator not real device. 
Because in real device we have Metamask app or other wallet app.
But in simulator we do not have any wallet.
For expo we need to build `development-client` and install to real device first.

**Comment by ignaciosantise** [comment]
@Kittinut-Muza what was the root cause of the issue? 

The SDK works on simulators and real devices
